src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java

```diff
@@ -99,23 +107,29 @@ public ComponentStore(NucleusPaths nucleusPaths, PlatformResolver platformResolv
      * @return persisted recipe content in component store on the disk.
      * @throws PackageLoadingException if fails to write the package recipe to disk.
      */
-    String saveComponentRecipe(@NonNull com.amazon.aws.iot.greengrass.component.common.ComponentRecipe componentRecipe)
+    String saveComponentRecipe(
+            @NonNull com.amazon.aws.iot.greengrass.component.common.ComponentRecipe componentRecipe)
             throws PackageLoadingException {
         ComponentIdentifier componentIdentifier =
-                new ComponentIdentifier(componentRecipe.getComponentName(), componentRecipe.getComponentVersion());
+                new ComponentIdentifier(
+                        componentRecipe.getComponentName(), componentRecipe.getComponentVersion());
 
         try {
             String recipeContent =
-                    com.amazon.aws.iot.greengrass.component.common.SerializerFactory.getRecipeSerializer()
+                    com.amazon.aws.iot.greengrass.component.common.SerializerFactory
+                            .getRecipeSerializer()
                             .writeValueAsString(componentRecipe);
 
-            Optional<String> componentRecipeContent = findComponentRecipeContent(componentIdentifier);
-            if (componentRecipeContent.isPresent() && componentRecipeContent.get().equals(recipeContent)) {
+            Optional<String> componentRecipeContent =
+                    findComponentRecipeContent(componentIdentifier);
+            if (componentRecipeContent.isPresent()
+                    && componentRecipeContent.get().equals(recipeContent)) {
                 // same content and no need to write again
                 return recipeContent;
             }
 
-            FileUtils.writeStringToFile(resolveRecipePath(componentIdentifier).toFile(), recipeContent);
+            FileUtils.writeStringToFile(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Path traversal vulnerability detected. User-controlled input in file paths
can allow attackers to access files outside intended directories using
`../` sequences. Secure your code by using framework functions like
`getCanonicalPath()`, `normalize()`, or validating paths with
`.startsWith()` checks. Learn more:
https://cwe.mitre.org/data/definitions/22.html